### PR TITLE
EventGrid: Swagger for preview API version 2018-05-01-preview

### PR DESCRIPTION
**Pull request by kalyanaj**
UPDATE 2: Actual Swagger to be reviewed for the 2018-05-01-preview. The diff between the second commit and third commit shows the changes requiring review in this new API version.

UPDATE 1: This is an initial placeholder version for a new preview API version (2018-05-01-preview) and to make future diffs easier, this is currently just an exact copy of the previous stable API version (2018-01-01) with the only change being changing API version to 2018-05-01-preview.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by kalyanaj** [comment]
@sarangan12 , will you be able to help review this today?  That will help us move forward in the pipeline (for SDKs and CLI/PowerShell) based on these features.  Thanks!

**Comment by kalyanaj** [comment]
Adding @ravbhatnagar as well - Gaurav, can you please help review this from the ARM perspective (more than 80% of this are areas you have already reviewed offline)?  There are no new APIs, this adds new definitions that will be additional properties on existing resource types.

**Review by sarangan12 — APPROVED** [review]

**Comment by sarangan12** [comment]
I am Ok with the changes. Waiting for @ravbhatnagar approval to merge

**Pull request merged**
